File: gateOne/Day Seven/java/MenstrualApp.java
 import java.util.Scanner; 
import java.time.LocalDate; 
import java.time.Period; 
import java.time.format.DateTimeFormatter; 

public class MenstrualApp { 

    public static void main(String[] args) { 
        Scanner input = new Scanner(System.in); 

        System.out.println("WELCOME TO OUR MENSTRUAL APP"); 

        System.out.println("Enter the day your period starts (dd/mm/yyyy):"); 
        String periodStart = input.nextLine(); 

        System.out.println("Enter the last day your period ends (dd/mm/yyyy):"); 
        String periodEnd = input.nextLine(); 

        System.out.println("How many days does your menstrual cycle length take?"); 
        int cycleLength = input.nextInt(); 

        LocalDate start = LocalDate.parse(periodStart, DateTimeFormatter.ofPattern("dd/MM/yyyy")); 
        LocalDate end = LocalDate.parse(periodEnd, DateTimeFormatter.ofPattern("dd/MM/yyyy")); 

        Period cyclePeriod = Period.between(start, end);

   	int ovulationDay = cycleLength / 2;
	int safeDays = (ovulationDay/2) ;

        LocalDate nextMenstrualCycleStart = end.plusDays(cycleLength);
        LocalDate nextMenstrualCycleEnd = nextMenstrualCycleStart.plus(cyclePeriod); 
        int cycleDuration = Period.between(nextMenstrualCycleStart, nextMenstrualCycleEnd).getDays(); 
	
	LocalDate nextOvulationDate = nextMenstrualCycleEnd.plusDays(ovulationDay);
	LocalDate nextOvulationEndDate = nextOvulationDate.plusDays(safeDays);
	LocalDate nextSafePeriodStartDate = nextOvulationDate.plusDays(1);

	LocalDate safePeriodBeforeNextMenstruationDate = nextMenstrualCycleStart.minusDays(safeDays);

	System.out.println("Safe period before next Mestruation Date : -"safePeriodBeforeNextMenstruationDate +" till next menstruation starts.");
        System.out.println("Start of next cycle: " + nextMenstrualCycleStart); 
        System.out.println("End of next cycle: " + nextMenstrualCycleEnd);

	System.out.println("First Safe Period after your next period: "+nextMenstrualCycleEnd +  " - " + nextMenstrualCycleEnd.plusDays(safeDays));

        System.out.println("Next Ovulation date : " + nextOvulationDate);

	System.out.println("Last safe Period after your next period: " + nextSafePeriodStartDate + " - " +  nextOvulationEndDate);
    }
}
 

File: gateOne/Day Seven/java Script/menstrualApp.js
const prompt = require('prompt-sync')();
const { format, parse, addDays, differenceInDays, add, sub, addYears } = require('date-fns');


const formatDate = (date) => format(date, 'dd/MM/yyyy');


const parseDate = (dateStr) => parse(dateStr, 'dd/MM/yyyy', new Date());

console.log("WELCOME TO OUR MENSTRUAL APP");

const periodStart = prompt("Enter the day your period starts (dd/mm/yyyy): ");
const periodEnd = prompt("Enter the last day your period ends (dd/mm/yyyy): ");
const cycleLength = parseInt(prompt("How many days does your menstrual cycle length take? ");

const start = parseDate(periodStart);
const end = parseDate(periodEnd);

const cyclePeriod = differenceInDays(end, start);
const ovulationDay = cycleLength / 2;
const safeDays = ovulationDay / 2;

const nextMenstrualCycleStart = addDays(end, cycleLength);
const nextMenstrualCycleEnd = addDays(nextMenstrualCycleStart, cyclePeriod);

const nextOvulationDate = addDays(nextMenstrualCycleEnd, ovulationDay);
const nextOvulationEndDate = addDays(nextOvulationDate, safeDays);
const nextSafePeriodStartDate = addDays(nextOvulationDate, 1);


console.log(`Start of next cycle: ${formatDate(nextMenstrualCycleStart)}`);
console.log(`End of next cycle: ${formatDate(nextMenstrualCycleEnd)}`);
console.log(`First Safe Period after your next period: ${formatDate(nextMenstrualCycleEnd)} - ${formatDate(addDays(nextMenstrualCycleEnd, safeDays))}`);
console.log(`Next Ovulation date: ${formatDate(nextOvulationDate)}`);
console.log(`Last safe Period after your next period: ${formatDate(nextSafePeriodStartDate)} - ${formatDate(nextOvulationEndDate)}`);
